### DISCUSSION TITLE
Can't reach database server

### DISCUSSION THREAD

**Discussion by MrJadaml** [supabase/supabase#12802] [Questions]
I keep running into the same issue where my application loses connection with the db after some time.

```
Error:  Invalid `prisma.business.findMany()` invocation:

Can't reach database server at `db.klbvssrmwmohmieumwsg.supabase.co`:`6543`
Please make sure your database server is running at `db.klbvssrmwmohmieumwsg.supabase.co`:`6543`.
```

```
# .env

DATABASE_URL="postgres://postgres:<MY_NO_SPECIAL_CHAR_PASSWORD>@db.klbvssrmwmohmieumwsg.supabase.co:6543/postgres"
```

```
// schema.prisma

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

...some models...
```

I've seen this issue in 3 different test apps I've created now, where it works fine at first, then after some time starts producing this error. In one of them after running into the issue again I created a new Project on Supabase and swapped out the connection url. That fixed things, at first, but then also started having the same issue again. Maybe I'm just unknowingly making the same mistake over and over?

Using Next.js + Prisma + Supabase
Following this guide: https://vercel.com/guides/nextjs-prisma-postgres

Repo here: https://github.com/MrJadaml/merchant-fox

**Comment by ilhamgepe** [comment]
i have same issue. i've try restart project in supabase. its work but this error randomly always showing. but i try this [solution](https://stackoverflow.com/a/73919700/16461906) adding "connection_timeout=300" in my env and work. i hope this help. 
eg: DATABASE_URL="postgres://postgres:xxxxx@db.xxxx.supabase.co:6543/postgres?pgbouncer=true&connect_timeout=300"

**Comment by souvik666** [comment]
hey @ilhamgepe  @MrJadaml it might be related to this [link](https://stackoverflow.com/questions/73866587/prisma-error-p1001-cant-reach-database-server-at-db-xocheossqzkirwnhzxxm-sup)

**Comment by benjamin-kirkbride** [comment]
I seem to be having the same issue.. intermittent database timeouts.

**Reply by isaachinman** [reply]
@benjamin-kirkbride Did you ever resolve the intermittent timeouts? This is exactly the behaviour I am seeing, random timeouts in our background worker after running fine for 2-3 hours.

**Comment by aidanmcintosh07** [comment]
What worked for me was adding `pgbouncer=true` to my database url. I used neon db and prisma but the same concept should apply.
Example:
`DATABASE_URL='postgresql://xxxxxxx:xxxxxxxx@xxxxxxx.xxx.xxx.neon.tech/xxxxxxx?sslmode=require&pgbouncer=true'`
